### PR TITLE
Replace GET_R3TR_OBJECT_FROM_LIMU_OBJ

### LINKED ISSUE #4715: Replace GET_R3TR_OBJECT_FROM_LIMU_OBJ
I noticed (in my own project) that this SAP function module does not behave as expected. Especially, for some subobject types, it does not return the main object, but the development class:

- DTED
- DOMD
- TABT
- MESS
- SOTT

I do not know what the effect is in ABAPGit (where it is used in ZCL_ABAPGIT_TRANSPORT->RESOLVE), but I suspect that it can lead to unwanted effects.
A good replacement seems to be function TR_CHECK_TYPE (just using parameters wi_e071 and we_tadir).

### PR DESCRIPTION
**Pull request by vonglan**
not tested in ABAPGit context, only locally.
First Pull Request, so please look closely.

### CODE REVIEW

**Comment by larshp** [comment]
#4715

File: src/cts/zcl_abapgit_transport.clas.abap [hunk 1]
```diff
@@ -240,20 +242,17 @@ CLASS zcl_abapgit_transport IMPLEMENTATION.
...
-              p_r3tr_objtype = lv_object
-              p_r3tr_objname = lv_trobj_name
-            EXCEPTIONS
-              no_mapping     = 1
-              OTHERS         = 2.
-          IF sy-subrc <> 0.
-            zcx_abapgit_exception=>raise( 'error from GET_R3TR_OBJECT_FROM_LIMU_OBJ' ).
+              we_tadir  = ls_tadir_sap
+              pe_result = lv_result.
+          IF 'TL' NS lv_result OR ls_tadir_sap IS INITIAL.
```
Comment (mbtools) [inline src/cts/zcl_abapgit_transport.clas.abap:251]: let's avoid yoda conditions

```abap
         IF lv_result NA 'TL' OR ls_tadir_sap IS INITIAL.
```

**Review by mbtools — CHANGES_REQUESTED** [review]
Thanks, Edo, for the fix. I tested your change and it works well in general. 

However, I discovered a bug that we should fix at this time. `rt_tadir` incorrectly contains objects that have been deleted already. For example:

image


Could you please change the `append` as follows?

```abap
        IF ls_tadir-delflag IS INITIAL.
          APPEND ls_tadir TO rt_tadir.
        ENDIF.
```

File: src/cts/zcl_abapgit_transport.clas.abap (hunk 1 above)
Comment (vonglan) [reply src/cts/zcl_abapgit_transport.clas.abap]: done

**Review by mbtools — APPROVED** [review]

**Comment by mbtools** [comment]
lgtm 👍

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Pull request merged**